idaes/models/unit_models/heat_exchanger_1D.py

```diff
@@ -430,39 +430,62 @@ def build(self):
 
         # Set flow directions for the control volume blocks and specify
         # discretization if not specified.
-        if self.config.flow_type == HeatExchangerFlowPattern.cocurrent:
-            set_direction_hot = FlowDirection.forward
-            set_direction_cold = FlowDirection.forward
+        if self.config.hot_side.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the hot side of the "
+                "co-current heat exchanger. "
+                "Defaulting to finite "
+                "difference method on the hot side."
+            )
+            self.config.hot_side.transformation_method = "dae.finite_difference"
+        if self.config.cold_side.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the cold side of the "
+                "co-current heat exchanger. "
+                "Defaulting to finite "
+                "difference method on the cold side."
+            )
+            self.config.cold_side.transformation_method = "dae.finite_difference"
+
+        if (
+            self.config.hot_side.transformation_method
+            != self.config.cold_side.transformation_method
+        ):
```

Comment:
In one of the tests, the `transformation_scheme` is different for the hot/cold sides and expected to throw a `ConfigurationError`, but it appears that the error is only thrown if the `transformation_method` is different for the hot/cold sides.

`@pytest.mark.unit
    def test_config_validation():
        m = ConcreteModel()
        m.fs = FlowsheetBlock(dynamic=False)
    
        m.fs.properties = BTXParameterBlock(valid_phase="Liq")
    
        with pytest.raises(ConfigurationError):
>           m.fs.HX_co_current = HX1D(
                hot_side={
                    "property_package": m.fs.properties,
                    "transformation_scheme": "BACKWARD",
                },
                cold_side={
                    "property_package": m.fs.properties,
                    "transformation_scheme": "FORWARD",
                },
                flow_type=HeatExchangerFlowPattern.cocurrent,
            )
E           Failed: DID NOT RAISE <class 'idaes.core.util.exceptions.ConfigurationError'>`

Reply:
The intention of the changes to the `HeatExchanger1D` was to allow the user to use an [upwind difference scheme](https://en.wikipedia.org/wiki/Upwind_scheme) for countercurrent flow and make that the default. Upwinding has some numerical benefits, but they may only apply to explicit time integration schemes. At the same time, using opposite schemes on the hot side/cold side causes energy to not be conserved. @andrewlee94  would say that's a result of the numerical approximation, and what did you expect using a first order method. Nevertheless, we need to think about what we want to do and whether introducing a change like this might cause problems for users.